File: 20211001-java-todo-list/src/test/java/com/katas/TodoListTest.java
package com.katas;

import org.junit.Test;

import static org.assertj.core.api.AssertionsForClassTypes.assertThat;

public class TodoListTest {

    private TodoList todoList;

    @Test
    public void total_duration_of_empty_list() {
        todoList = new TodoList();

        assertThat(todoList.totalDuration()).isZero();
    }

    @Test
    public void total_duration_of_list_with_one_task() {
        todoList = new TodoList(aTaskWithDuration(5));

        assertThat(todoList.totalDuration()).isEqualTo(5);
    }

    @Test
    public void total_duration_of_list_with_multiple_tasks() {
        todoList = new TodoList(aTaskWithDuration(3), aTaskWithDuration(5));

        assertThat(todoList.totalDuration()).isEqualTo(8);
    }

    private Task aTaskWithDuration(int duration) {
        return new Task(duration);
    }
}

File: 20211001-kotlin-todo-list/src/test/kotlin/com/katas/TodoListTest.kt
package com.katas

import org.assertj.core.api.AssertionsForClassTypes.assertThat
import org.junit.jupiter.api.Test

class TodoListTest {
    @Test
    fun `total duration of empty list`() {
        val todoList = TodoList(listOf())

        assertThat(todoList.totalDuration()).isZero
    }

    @Test
    fun `total duration when one item`() {
        val todoList = TodoList(listOf(Task(duration = 5)))

        assertThat(todoList.totalDuration()).isEqualTo(5)
    }

    @Test
    fun `total duration when multiple items`() {
        val todoList = TodoList(listOf(Task(duration = 5), Task(duration = 3)))

        assertThat(todoList.totalDuration()).isEqualTo(8)
    }
}
